lucene/core/src/java/org/apache/lucene/search/TopScoreDocCollector.java

```diff
@@ -155,6 +175,11 @@ public void collect(int doc) throws IOException {
           pqTop.doc = doc + docBase;
           pqTop.score = score;
           pqTop = pq.updateTop();
+
+          if (bottomValueChecker != null && bottomValueChecker.getBottomValue() > 0) {
```

Comment:
The update below should be called only when the total hits threshold is used and the queue is full so it would be easier to set it in `updateMinCompetitiveScore` like we do to propagate the min score in the scorer ?